### ISSUE TITLE
Some API resources does not return any data on POST

### ISSUE THREAD

**Issue by emjemj** [NORDUnet/ni#1]
When you POST to create a resource, some resources return the created object in JSON format and some return nothing at all. For instance Cable returns data and Site does not. 

Is this by design or is the API just not used very much?

**Comment by mkrogh** [comment]
Well the api is not used much, and as such is most exercised by the testcases (which of cause are not full coverage).

I will have a look at it shortly, and see what is up.

**Comment by emjemj** [comment]
I figured that was the case. It seems to be related to the obj_create method in NodeHandleResource. I looked at it and didn't see anything obvious.

**Comment by mkrogh** [comment]
So a quick look later and it seems like it is default for tastypie to return a `201` which means "created". This is accompanied by a Location header that points to the created resource e.g. `/api/v1/site/handle_id`.

But for some of the resources, such as `Cable` we actually as you say return the data that was created, along with the `Location` header. But the location header is wrong it uses the `node_name` not the handle id... so something is indeed weird with the `Cable` resource.

I am leaning mostly at what the rest of the models do: `201` and just `Location` header. But I might willing to listen to reason if it is more convenient that it returns the full object. It should however still return the correct uri's.

**Comment by mkrogh** [comment]
The `Cable` resource is using the node_name due to NDN using unique cable ids, so the location header makes sense in some way. (I still think that it should handle both `handle_id` and `node_name`)

**Comment by emjemj** [comment]
I see that's usable. I'm thinking about building a python lib for it, to automate boring parts. Knowing that the Location header is there makes things a lot easier. Thanks! 

I know all about the nordunet specifics of this system. A bit of a pain in the neck when you're not nordunet. ;)

**Comment by mkrogh** [comment]
Yeah, though the newest version at least tries to dial the branding back a bit :)

I am going on vacation to day, so wont be doing any changes for this for the next 3 weeks, but glad you at least got part of the problem solved, in that you can code around using the `Location` header in the response. That should be there in all of the responses to create.

I have a couple of changes in the pipeline, such as not having to set the `node_type` and `node_meta_type` when you do a `POST` to a specific endpoint such as `/api/v1/site/`, since what you are posting is probably a `site` resource, and those are a `Location` meta type. This improvement might hit the repo today, if I can figure out a neat way to do it. If not then after my vacation.
